setuptools/extension.py

```diff
@@ -27,7 +27,7 @@ def _have_cython():
 _Extension = get_unpatched(distutils.core.Extension)
 
 
-class Extension(_Extension):
+class Extension(_Extension):  # type: ignore[valid-type, misc]  # https://github.com/python/mypy/issues/14458
```

Comment:
Would it improve things if we wrap the `get_unpatched` into the `else` branch of an `if TYPE_CHECKING`?

Reply:
Yes (same as above, that's also all 3 usages of get_unpatched)